ocaml/testsuite/tests/typing-implicit-source-positions/invalid_usages.ml

```diff
@@ -44,7 +44,7 @@ Line 1, characters 14-15:
                   ^
 Error: This expression has type call_pos:[%call_pos] -> unit -> unit
        but an expression was expected of type
-         call_pos:Lexing.position -> (unit -> 'a)
+         call_pos:Lexing.position -> unit -> 'a
```

Comment:
It seems that this PR improves the error message here: the root cause here is type mismatch (`[%call_pos]` vs `Lexing.position`), hence the extra parens is just confusing and shouldn't be printed.